qiita_pet/templates/search_sample_list.html

```diff
@@ -0,0 +1,65 @@
+<!--Study datatypes tabs-->
+{% from future.utils import viewitems %}
+{% from qiita_pet.util import clean_str %}
+{% set sel = "" %}
+{% if selected is True %}
+    {% set sel = "-sel" %}
+  {% end %}
+
+<ul class="nav nav-tabs" role="tablist" id="dt-tabs-{{sid}}{{sel}}">
+{% for data_type in proc_data_dict %}
+  <li class="active"><a href="#{{sid}}-{{data_type}}{{sel}}" role="tab" data-toggle="tab">{{data_type}}</a></li>
+{% end %}
+</ul>
+
+{% for data_type, proc_data in viewitems(proc_data_dict) %}
+<div class="tab-content" id="pane-{{sid}}-{{data_type}}{{sel}}">
+  <!--Study datatype processed data tabs-->
+  <ul class="nav nav-tabs" role="tablist" id="dt-tabs-{{sid}}-proc-{{sel}}">
+  {% for pid in proc_data %}
+    <li class="active"><a href="#pane-{{sid}}-{{data_type}}-{{pid}}{{sel}}" role="tab" data-toggle="tab">Proc Data {{pid}}</a></li>
+  {% end %}
+  </ul>
+{% end %}
+
+<!--Sample divs per datatype per processed data id-->
+{% for data_type, proc_data in viewitems(proc_data_dict) %}
+  {% for pid in proc_data %}
+  <div class="tab-content" id="pane-{{sid}}-{{data_type}}-{{pid}}{{sel}}">
+    <!--samples list-->
+    <h3>Samples</h3>
+    <a href="#" onclick="select_deselect({{sid}}, {{pid}}, '{{sel}}', true); return false;">Select all</a> | <a href="#" onclick="select_deselect({{sid}}, {{pid}}, '{{sel}}', false); return false;">Select none</a> | <a href="#" onclick="select_inverse({{sid}}, {{pid}}, '{{sel}}'); return false;">Select inverse</a>
+  <table class='table table-hover'>
+    <tr>
+      <th></th>
+      <th></th>
+      <th>Sample ID</th>  
+    {% for meta in meta_headers %}
+      <th>{{meta}}</th>
+    {% end %}
+    </tr>
+    {% if meta_headers %}
+      {% for sample in proc_data_samples[pid]%}
+    <tr><td>
+      <input type='checkbox' name='samples{{sel}}'
+      value='{{pid}}#{{sample[0]}}' class='study{{sid}} proc{{pid}} {{ " ".join(clean_str(s) for s in sample) }} {{sel}}' onchange='count_update({{sid}})'></td>
+      <td><a href="#" onclick="select_category('{{clean_str(sample[0])}}', '')">All Datatypes</a></td>
+        {% for val in sample %}
```

Comment:
also, if I introduce another object into the fray, I can cut down on the regex calls. This would mean there is a dict of lists with proc_data_ids keyed to their studies, a dict of lists with each sample id keyed to the proc_data_id it is in, and new third that is a dict of list of tuple, each tuple containing the samples with metadata, and each list of samples keyed to the metadata. This means we can run through the metadata cleaning once for all samples instead of each time it is printed, which could be multiple times per study.